node/hyperbahn/handler.js

```diff
@@ -294,5 +329,12 @@ function logError(err, opts, response) {
 
 HyperbahnHandler.prototype.advertise =
 function advertise(service) {
-    throw new Error('not implemented');
+    var self = this;
+    self.channel.topChannel.handler.refreshServicePeer(service.serviceName, service.hostPort);
```

Comment:
This is not ideal; Would be better to do `self.serviceProxy.refreshServicePeer()`